Ska/ParseCM.py

```diff
@@ -172,4 +173,37 @@ def read_mm(filename):
     return manvr_list
 
 
+def read_si_align(char_file):
+    """
+    Read ODB_SI_ALIGN matrices from supplied characteristics file
 
+    :param char_file: OFLS ODB characteristics file
+    :returns: dictionary of matrices, one per detector in C/Python order
+    """
+    char_lines = open(char_file).readlines()
+    # Find the beginning of the assignment using the regex from aimpoint_mon
+    RE_ODB_SI_ALIGN = re.compile(r'\s* ODB_SI_ALIGN \s* =', re.VERBOSE)
+    matches = [i for i, line in enumerate(char_lines) if RE_ODB_SI_ALIGN.match(line)]
+    if len(matches) != 1:
+        raise ValueError('parsing error, matched {} instances of ODB_SI_ALIGN instead of one'
+                         .format(len(matches)))
+    start = matches[0]
+    # Grab the 12 lines of the alignment.  This assumes no extra blank lines
+    odb_si_lines = char_lines[start:start+12]
+    # Remove ODB_SI_ALIGN label
+    odb_si_lines[0] = re.sub(RE_ODB_SI_ALIGN, "", odb_si_lines[0])
+    # Remove comments that begin with an exclamation mark
+    odb_si_lines = [re.sub("!.*$", "", line) for line in odb_si_lines]
+    # Remove leading and trailing whitespace
+    odb_si_lines = [line.strip() for line in odb_si_lines]
+    # Matching the values using a simple comma/space pattern with very little checking
+    odb_strings = [list(re.search("(\S+)\s* , \s*(\S+)\s* , \s*(\S+)\s* ,", line, re.VERBOSE).groups())
```

Comment:
I suppose instead of "\S" I should really match non-whitespace character that isn't a comma.